src/main/java/io/mosip/iiitb/odte/consumers/OnDemandTemplateExtractionConsumerImpl.java

```diff
@@ -0,0 +1,198 @@
+
+        try {
+            credentialRequestAdditionalData = this.getAdditionalData(
+                    uid,
+                    record.getIdType(),
+                    tokenId
+            );
+        } catch (JsonProcessingException | NoSuchAlgorithmException ex) {
+            logger.error(ex.getMessage());
+            return eventConsumerResponse(EventConsumerStatus.ERROR);
+        }
+
+        logger.debug("additional data = {}}\n", credentialRequestAdditionalData);
+
+        String requestId = null;
+        try {
+            requestId = apiRequestService.getCredentialRequestId(
+                    authToken,
+                    uid,
+                    issueCredentialsAuthConf.getIssuer(),
+                    credentialRequestAdditionalData
+            );
+            logger.debug("request id = {}\n", requestId);
+        } catch (IOException | InterruptedException ex) {
+            logger.error("Failed to get requestId");
+            logger.error(ex.getMessage());
+            return eventConsumerResponse(EventConsumerStatus.ERROR);
+        }
+
+        String issueCredAuthToken = null;
+        try {
+            issueCredAuthToken = apiRequestService.getAuthToken(
+                    issueCredentialsAuthConf.getAppId(),
+                    issueCredentialsAuthConf.getClientId(),
+                    issueCredentialsAuthConf.getClientPass()
+            );
+        }  catch (IOException | InterruptedException ex) {
+            logger.error("Failed to get issueCredAuthToken ");
+            logger.error(ex.getMessage());
+            return eventConsumerResponse(EventConsumerStatus.ERROR);
+        }
+
+        String issueStatus = null;
+        try {
+            IssueCredentialsRawResponseDto rawResponse = apiRequestService.issueCredentials(
+                    issueCredAuthToken,
+                    uid,
+                    partnerCode,
+                    requestId,
+                    credentialRequestAdditionalData
+            );
+            this.logger.debug(
+                String.format("Issue credential response = %s", rawResponse)
+            );
+            MosipResponseError[] errors = rawResponse.getErrors();
+            if (errors != null && errors.length > 0) {
+                logger.error("Failed To Issue The given credentials.");
+                for (MosipResponseError error: errors) {
+                    logger.error(
+                            String.format("%s", error)
+                    );
+                }
+            }
+            IssueCredentialsResponseDto response = rawResponse.getResponse();
+            if (response != null) {
+                issueStatus = response.getStatus();
+            }
+        } catch (Exception e) {
+            logger.error("Error = " + e);
+            return eventConsumerResponse(EventConsumerStatus.ERROR);
+        }
+        logger.debug("Status = " + issueStatus);
+        return eventConsumerResponse(EventConsumerStatus.SUCCESS);
```

Comment:
This method looks very long, can we break it into multiple smaller methods and call ? 
Also if we are anyway catching exception and returning, why not have everything under a single try, catch once and return ?